platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/PropertyExtensions.kt

```diff
@@ -103,3 +104,138 @@ fun <K, V> MapProperty<K, V>.assign(entries: Map<out K?, V?>?) {
 fun <K, V> MapProperty<K, V>.assign(provider: Provider<out Map<out K?, V?>?>) {
     this.set(provider)
 }
+
+
+/**
+ * Adds an element to the property value.
+ *
+ * @see HasMultipleValues.add
+ * @since 9.0
+ */
+@Incubating
+operator fun <T : Any> HasMultipleValues<T>.plusAssign(element: T) {
+    // TODO(mlopatkin): consider changing this to a function with better incremental update semantics
+    this.add(element)
```

Comment:
❓  Would be ok to cast here to `CollectionPropertyInternal` and use `append` already now?

Reply:
> ❓ Would be ok to cast here to `CollectionPropertyInternal` and use `append` already now?
> 
> And similar for other cases.

I don't think it is a good idea, `append` for now shows behavior we're unlikely to push forward (of committing the convention eagerly)